libs/utils/energy_model.py

```diff
@@ -1001,6 +1025,16 @@ def read_idle_states(cpu, domain_level):
                    root_power_domain=root_pd,
                    freq_domains=freq_domains)
 
+    @staticmethod
+    def save_em_yaml(em, yaml_path):
```

Comment:
Naming nit: Could those be named `to_yaml` and `from_yaml` ? That would better match what we currently have (`from_target`). Also those methods are under the EnergyModel class, so IMO there's no need to specify "em" in the method name.

Also, `from_yaml` should be a classmethod, and `to_yaml` can be a regular non-static method (or a classmethod if you want some sort of symmetry, but I'd find it more cumbersome than anything else).

Those are comments that I could have made way back then, but they only hit me as I started testing this PR...

Reply:
Hm the names I suggested interfere with `ruamel.yaml` and how it serializes objects. I guess we can use something like `to/from_disk` instead, and it's even neater because we don't expose what file format we're using :D

Reply:
Unfortunately, these names are used by ruamel.yam:
https://yaml.readthedocs.io/en/latest/dumpcls.html

Ultimately, these helpers should go in a base class so it can be used for other classes as well, so it may not be ready to merge yet.

Reply:
Aye, a base class for those would be nice.